source/js/nav-newsletter.js

```diff
@@ -0,0 +1,162 @@
+import React from "react";
+import ReactGA from "react-ga";
+import ReactDOM from "react-dom";
+import JoinUs from "./components/join/join.jsx";
+
+let primaryNav,
```

Comment:
Let's change this to an object, due to the fact that we can make checkDomNodes a lot smaller:
```javascript
const elements = {
  primaryNav: `#primary-nav-container`,
  narrowMenuContainer: `.wide-screen-menu-container`,
  wideMenuContainer: ...,
  container: ...,
  buttonDesktop: ...,
  buttonMobile: ...,
  joinUs: ...,
  buttonDismiss: ...
};
```
With corresponding changes so that `container..` becomes `elements.container` etc.